src/sparseml/pytorch/torchvision/train.py

```diff
@@ -561,9 +561,17 @@ def log_metrics(tag: str, metrics: utils.MetricLogger, epoch: int, epoch_step: i
             loggers=logger,
             distillation_teacher=args.distill_teacher,
         )
-        optimizer = manager.modify(
-            model, optimizer, steps_per_epoch=steps_per_epoch, epoch=args.start_epoch
+        step_wrapper = manager.modify(
+            model,
+            optimizer,
+            steps_per_epoch=steps_per_epoch,
+            epoch=args.start_epoch,
+            wrap_optim=scaler,
         )
+        if scaler is None:
+            optimizer = step_wrapper
+        else:
+            scaler = step_wrapper
```

Comment:
Will re-ask this question here:

If using quantization & amp, the scaler will be set to None down below. This means that train_one_epoch will directly using optimizer instead of scaler. However the manager.modify wrapped the scaler.

Does that mean I need to re-call manager.modify on the optimizer after amp is disabled?

Reply:
Or should I disable scaler by doing `scaler._enabled = False`? That seems to be what transformers integration does

Reply:
Resolved in meeting - setting enabled flag for scaler down below

Reply:
As mentioned offline, just setting `scaler._enabled = False` is enough. The optimizer step will be taken care of by torch here:
https://github.com/pytorch/pytorch/blob/d8e795ecd53670682bd3b2e5ff1f378402b147d5/torch/cuda/amp/grad_scaler.py#L313